include/fiction/algorithms/physical_design/post_layout_optimization.hpp

```diff
@@ -371,20 +404,14 @@ layout_coordinate_path<Lyt> get_path_and_obstruct(Lyt& lyt, const tile<Lyt>& sta
  * @tparam Lyt Cartesian obstruction gate-level layout type.
  * @param lyt 2DDWave-clocked cartesian obstruction gate-level layout.
  * @param old_pos Old position of the gate to be moved.
- * @return Flag that indicates if gate was moved successfully and the new coordinate of the moved gate.
+ * @return `true` if the gate was moved successfully, `false` otherwise.
  */
 template <typename Lyt>
-[[nodiscard]] std::pair<bool, tile<Lyt>> improve_gate_location(Lyt& lyt, const tile<Lyt>& old_pos) noexcept
+[[nodiscard]] bool improve_gate_location(Lyt& lyt, const tile<Lyt>& old_pos, const tile<Lyt>& max_non_po,
```

Comment:
Is it really an error if the return type is not utilized here? The gate location is improved either way, right? I would argue that this function should not be `[[nodiscard]]`